### ISSUE TITLE
[No Data] Z3 data for RESIF stations not available

### ISSUE THREAD

**Issue by jbienkowski** [EIDA/userfeedback#35]
Issue created on behalf of one of our users:

> Good morning,
> 
> I am in the AlpArray core group (got authentication and tocken). I am trying to download (webservices)some data for a specific period in 2017 but I cannot access all the RESIF stations. Could you please advice?
> 
> Thanks and best regards, Andrea
> 
> My query is:
> 
> fdsnws_fetch -a eidatoken -N _ALPARRAY -S '*' -C HHZ -L -- -s 2017-08-23T07:29:00 -e 2017-08-23T07:39:00 -v -o 2017-08-23T07:29:00_2017-08-23T07:39:00.mseed
> 
> Here below the output, authentication looks ok but I do not get data
> 
> Thanks for checking. Andrea

```
'using token in eidatoken:
      [31m[Errno 2] No such file or directory: 'gpg': 'gpg' [m
     getting routes from http://geofon.gfz-potsdam.de/eidaws/routing/1/query?network=_ALPARRAY&station=%2A&channel=HHZ&location=--&starttime=2017-08-23T07%3A29%3A00&endtime=2017-08-23T07%3A39%3A00&format=post
      [31mdid not receive routes to _ALPARRAY.*..HHZ [m
     authenticating at https://eida.ethz.ch/fdsnws/dataselect/1/auth
     authenticating at https://ws.resif.fr/fdsnws/dataselect/1/auth
     authenticating at https://erde.geophysik.uni-muenchen.de/fdsnws/dataselect/1/auth
     authenticating at https://www.orfeus-eu.org/fdsnws/dataselect/1/auth
     authentication at https://eida.ethz.ch/fdsnws/dataselect/1/auth successful
     getting data from http://eida.ethz.ch/fdsnws/dataselect/1/queryauth
     authentication at https://ws.resif.fr/fdsnws/dataselect/1/auth successful
     getting data from http://ws.resif.fr/fdsnws/dataselect/1/queryauth
     authentication at https://erde.geophysik.uni-muenchen.de/fdsnws/dataselect/1/auth successful
     getting data from http://erde.geophysik.uni-muenchen.de/fdsnws/dataselect/1/queryauth
     authentication at https://www.orfeus-eu.org/fdsnws/dataselect/1/auth successful
     getting data from http://www.orfeus-eu.org/fdsnws/dataselect/1/queryauth
     authenticating at https://webservices.ingv.it/fdsnws/dataselect/1/auth
     authentication at https://webservices.ingv.it/fdsnws/dataselect/1/auth successful
     getting data from http://webservices.ingv.it/fdsnws/dataselect/1/queryauth
     got 7539712 bytes (application/vnd.fdsn.mseed) from http://eida.ethz.ch/fdsnws/dataselect/1/queryauth
     authenticating at https://geofon.gfz-potsdam.de/fdsnws/dataselect/1/auth
     authentication at https://geofon.gfz-potsdam.de/fdsnws/dataselect/1/auth successful
     getting data from http://geofon.gfz-potsdam.de/fdsnws/dataselect/1/queryauth
      [31mreceived no data from http://ws.resif.fr/fdsnws/dataselect/1/queryauth [m
     authenticating at https://eida.bgr.de/fdsnws/dataselect/1/auth
     authentication at https://eida.bgr.de/fdsnws/dataselect/1/auth successful
     getting data from http://eida.bgr.de/fdsnws/dataselect/1/queryauth
     got 6819840 bytes (application/vnd.fdsn.mseed) from http://erde.geophysik.uni-muenchen.de/fdsnws/dataselect/1/queryauth
     got 1541120 bytes (application/vnd.fdsn.mseed) from http://eida.bgr.de/fdsnws/dataselect/1/queryauth
     got 1619456 bytes (application/vnd.fdsn.mseed) from http://geofon.gfz-potsdam.de/fdsnws/dataselect/1/queryauth
     got 9316352 bytes (application/vnd.fdsn.mseed) from http://webservices.ingv.it/fdsnws/dataselect/1/queryauth
     got 2509312 bytes (application/vnd.fdsn.mseed) from http://www.orfeus-eu.org/fdsnws/dataselect/1/queryauth
      [31mdid not receive data from BW.MANZ..HHZ, CH.BALST..HHZ, CH.OTER1..HHZ, CH.ROMAN..HHZ, CH.SALAN..HHZ, CH.SLE..HHZ, CH.WOLEN..HHZ, CR.HVAR..HHZ, CR.KALN..HHZ, CR.KIJV..HHZ, CR.MORI..HHZ, CR.MOSL..HHZ, CR.NVLJ..HHZ, CR.OZLJ..HHZ, CR.PLIT..HHZ, CR.PTJ..HHZ, CR.RABC..HHZ, CR.RIY..HHZ, CR.SMRN..HHZ, CR.UDBI..HHZ, CR.VIRC..HHZ, CR.ZAG..HHZ, CZ.PBCC..HHZ, CZ.PRA..HHZ, FR.AGO..HHZ, FR.AJAC..HHZ, FR.ARBF..HHZ, FR.ARTF..HHZ, FR.ASEAF..HHZ, FR.BANN..HHZ, FR.BETS..HHZ, FR.BLAF..HHZ, FR.BOUC..HHZ, FR.BSTF..HHZ, FR.CALF..HHZ, FR.CHMF..HHZ, FR.CIEL..HHZ, FR.CLAF..HHZ, FR.COLF..HHZ, FR.CORF..HHZ, FR.EILF..HHZ, FR.ENAUX..HHZ, FR.ESCA..HHZ, FR.FLAF..HHZ, FR.FRNF..HHZ, FR.GRN..HHZ, FR.HOHE..HHZ, FR.ILLF..HHZ, FR.ILLK..HHZ, FR.ISO..HHZ, FR.LAJAS..HHZ, FR.LBL..HHZ, FR.LEMB..HHZ, FR.MLYF..HHZ, FR.MOF..HHZ, FR.MON..HHZ, FR.MORSI..HHZ, FR.MVIF..HHZ, FR.NEEW..HHZ, FR.OG35..HHZ, FR.OGAG..HHZ, FR.OGCB..HHZ, FR.OGCC..HHZ, FR.OGDI..HHZ, FR.OGGM..HHZ, FR.OGLC..HHZ, FR.OGMO..HHZ, FR.OGMY..HHZ, FR.OGRV..HHZ
 , FR.OGS1..HHZ, FR.OGS2..HHZ, FR.OGS3..HHZ, FR.OGSA..HHZ, FR.OGSI..HHZ, FR.OGSM..HHZ, FR.OGVG..HHZ, FR.OPS..HHZ, FR.PIAF..HHZ, FR.PLDF..HHZ, FR.PLYF..HHZ, FR.PRIMA..HHZ, FR.RIVEL..HHZ, FR.RONF..HHZ, FR.RSL..HHZ, FR.RUSF..HHZ, FR.SALSA..HHZ, FR.SAOF..HHZ, FR.SAUF..HHZ, FR.SMPL..HHZ, FR.SPIF..HHZ, FR.STR..HHZ, FR.SURF..HHZ, FR.SZBH..HHZ, FR.TRBF..HHZ, FR.TRIGF..HHZ, FR.TURF..HHZ, FR.VAL4..HHZ, FR.VDH4..HHZ, FR.VERF..HHZ, FR.VOEL..HHZ, FR.WALT..HHZ, FR.WLS..HHZ, FR.ZELS..HHZ, GR.GEB2..HHZ, GR.GED1..HHZ, GR.GED5..HHZ, GU.GORR..HHZ, GU.MAIM..HHZ, GU.RRL..HHZ, HU.MPLH..HHZ, IV.ASQU..HHZ, IV.FSSB..HHZ, IV.MAGA..HHZ, IV.MAGO..HHZ, IV.OPPE..HHZ, MN.TRI..HHZ, MT.AVM..HHZ, MT.AVP..HHZ, MT.AVR..HHZ, MT.GAR..HHZ, MT.RES..HHZ, NI.PALA..HHZ, NI.VINO..HHZ, OE.BIOA..HHZ, OE.LESA..HHZ, OE.RONA..HHZ, OE.VIE..HHZ, OX.FERB..HHZ, RD.LOR..HHZ, RD.ORIF..HHZ, RD.PGF..HHZ, SI.ABSI..HHZ, SI.KOSI..HHZ, SL.BOJS..HHZ, SL.CADS..HHZ, SL.CEY..HHZ, SL.CRES..HHZ, SL.CRNS..HHZ, SL.DOBS..HHZ, SL.GBAS..HHZ, SL.GBRS..HHZ
 , SL.GCIS..HHZ, SL.GOLS..HHZ, SL.GORS..HHZ, SL.GROS..HHZ, SL.JAVS..HHZ, SL.KNDS..HHZ, SL.KOGS..HHZ, SL.LEGS..HHZ, SL.LJU..HHZ, SL.MOZS..HHZ, SL.PDKS..HHZ, SL.PERS..HHZ, SL.ROBS..HHZ, SL.SKDS..HHZ, SL.VISS..HHZ, SL.VNDS..HHZ, SL.VOJS..HHZ, SL.ZALS..HHZ, SL.ZAVS..HHZ, ST.GAGG..HHZ, Z3.A001A..HHZ, Z3.A002B..HHZ, Z3.A003A..HHZ, Z3.A004A..HHZ, Z3.A005B..HHZ, Z3.A006A..HHZ, Z3.A007A..HHZ, Z3.A008A..HHZ, Z3.A009A..HHZ, Z3.A010A..HHZ, Z3.A011B..HHZ, Z3.A012A..HHZ, Z3.A013A..HHZ, Z3.A014A..HHZ, Z3.A016A..HHZ, Z3.A017A..HHZ, Z3.A018A..HHZ, Z3.A019A..HHZ, Z3.A020A..HHZ, Z3.A021A..HHZ, Z3.A024B..HHZ, Z3.A030A..HHZ, Z3.A032A..HHZ, Z3.A033A..HHZ, Z3.A035A..HHZ, Z3.A036A..HHZ, Z3.A037A..HHZ, Z3.A038A..HHZ, Z3.A088B..HHZ, Z3.A121A..HHZ, Z3.A122A..HHZ, Z3.A123A..HHZ, Z3.A124A..HHZ, Z3.A125A..HHZ, Z3.A126A..HHZ, Z3.A127A..HHZ, Z3.A128A..HHZ, Z3.A129A..HHZ, Z3.A140A..HHZ, Z3.A141A..HHZ, Z3.A143A..HHZ, Z3.A144A..HHZ, Z3.A145A..HHZ, Z3.A146A..HHZ, Z3.A147B..HHZ, Z3.A148B..HHZ, Z3.A151A..HHZ, Z3.A152A..H
 HZ, Z3.A153A..HHZ, Z3.A154A..HHZ, Z3.A155A..HHZ, Z3.A156A..HHZ, Z3.A157B..HHZ, Z3.A158A..HHZ, Z3.A159A..HHZ, Z3.A160A..HHZ, Z3.A161A..HHZ, Z3.A162A..HHZ, Z3.A163A..HHZ, Z3.A164A..HHZ, Z3.A165A..HHZ, Z3.A166A..HHZ, Z3.A167A..HHZ, Z3.A168A..HHZ, Z3.A169A..HHZ, Z3.A170A..HHZ, Z3.A171A..HHZ, Z3.A172A..HHZ, Z3.A173A..HHZ, Z3.A174A..HHZ, Z3.A175A..HHZ, Z3.A176A..HHZ, Z3.A177A..HHZ, Z3.A178A..HHZ, Z3.A179A..HHZ, Z3.A180A..HHZ, Z3.A181A..HHZ, Z3.A182A..HHZ, Z3.A183A..HHZ, Z3.A184A..HHZ, Z3.A185A..HHZ, Z3.A186B..HHZ, Z3.A187A..HHZ, Z3.A188A..HHZ, Z3.A190A..HHZ, Z3.A191A..HHZ, Z3.A192A..HHZ, Z3.A193A..HHZ, Z3.A194A..HHZ, Z3.A195A..HHZ, Z3.A196A..HHZ, Z3.A198A..HHZ, Z3.A199A..HHZ, Z3.A200A..HHZ, Z3.A201A..HHZ, Z3.A202A..HHZ, Z3.A204A..HHZ, Z3.A205A..HHZ, Z3.A206A..HHZ, Z3.A208A..HHZ, Z3.A209B..HHZ, Z3.A210A..HHZ, Z3.A211A..HHZ, Z3.A212A..HHZ, Z3.A213A..HHZ, Z3.A214A..HHZ, Z3.A215A..HHZ, Z3.A216A..HHZ, Z3.A217A..HHZ, Z3.A252A..HHZ, Z3.A260A..HHZ, Z3.A261A..HHZ, Z3.A262A..HHZ, Z3.A263A..HHZ, Z3.
 A264A..HHZ, Z3.A265A..HHZ, Z3.A266A..HHZ, Z3.A267A..HHZ, Z3.A268A..HHZ, Z3.A269A..HHZ, Z3.A270A..HHZ, Z3.A288A..HHZ, Z3.A302A..HHZ, Z3.A305A..HHZ, Z3.A316A..HHZ, Z3.A331A..HHZ, Z3.A332A..HHZ, Z3.A333A..HHZ, Z3.A334A..HHZ, Z3.A335A..HHZ, Z3.A336A..HHZ, Z3.A337A..HHZ, Z3.A338A..HHZ, Z3.A339A..HHZ, Z3.A351A..HHZ, Z3.A358A..HHZ, Z3.A359A..HHZ, Z3.A368A..HHZ, Z3.A401A..HHZ, Z3.A410A..HHZ, Z3.A413A..HHZ, Z3.A416A..HHZ, Z3.A419A..HHZ, Z3.A422A..HHZ, Z3.A425A..HHZ, Z3.A429A..HHZ, ZS.D015..HHZ, ZS.D034..HHZ, ZS.D035..HHZ, ZS.D053..HHZ, ZS.D054..HHZ, ZS.D055..HHZ, ZS.D056..HHZ, ZS.D059..HHZ, ZS.D060..HHZ, ZS.D071..HHZ, ZS.D072..HHZ, ZS.D073..HHZ, ZS.D074..HHZ, ZS.D075..HHZ, ZS.D076..HHZ, ZS.D078..HHZ, ZS.D079..HHZ, ZS.D092..HHZ, ZS.D093..HHZ, ZS.D094..HHZ, ZS.D095..HHZ, ZS.D096..HHZ, ZS.D097..HHZ, ZS.D098..HHZ, ZS.D099..HHZ, ZS.D112..HHZ, ZS.D113..HHZ, ZS.D114..HHZ, ZS.D115..HHZ, ZS.D116..HHZ, ZS.D117..HHZ, ZS.D118..HHZ, ZS.D119..HHZ, ZS.D120..HHZ, ZS.D133..HHZ, ZS.D134..HHZ, ZS.D135..HHZ, ZS
 .D136..HHZ, ZS.D137..HHZ, ZS.D139..HHZ [m
     retrieving network citation info
     getting routes from http://geofon.gfz-potsdam.de/eidaws/routing/1/query
     getting data from http://geofon.gfz-potsdam.de/fdsnws/station/1/query
     getting data from http://webservices.ingv.it/fdsnws/station/1/query
     getting data from http://eida.bgr.de/fdsnws/station/1/query
     getting data from http://eida.ethz.ch/fdsnws/station/1/query
     getting data from http://erde.geophysik.uni-muenchen.de/fdsnws/station/1/query
     got 267 bytes (text/plain) from http://eida.ethz.ch/fdsnws/station/1/query
     getting data from http://www.orfeus-eu.org/fdsnws/station/1/query
     got 127 bytes (text/plain) from http://eida.bgr.de/fdsnws/station/1/query
     getting data from http://ws.resif.fr/fdsnws/station/1/query
     got 191 bytes (text/plain) from http://erde.geophysik.uni-muenchen.de/fdsnws/station/1/query
     got 312 bytes (text/plain) from http://www.orfeus-eu.org/fdsnws/station/1/query
     got 136 bytes (text/plain) from http://ws.resif.fr/fdsnws/station/1/query
     got 657 bytes (text/plain) from http://webservices.ingv.it/fdsnws/station/1/query
     got 348 bytes (text/plain) from http://geofon.gfz-potsdam.de/fdsnws/station/1/query
      [32m
     You received seismic waveform data from the following network(s): [m
      [32mBW BayernNetz [m
      [32mC4 CERN Seismic Network [m
      [32mCH National Seismic Networks of Switzerland [m
      [32mCR CR network [m
      [32mCZ Czech Regional Seismic Network [m
      [32mGE GEOFON Program, GFZ Potsdam, Germany [m
      [32mGR German Regional Seismic Network, BGR Hannover [m
      [32mGU Regional Seismic Network of North Western Italy [m
      [32mHU Hungarian National Seismological Network, MTA CSFK GGI (GGI_Net) [m
      [32mIU (GSN) IRIS/USGS (IU), GEOFON (GE), and GEUS [m
      [32mIV Italian Seismic Network [m
      [32mMN Mediterranean Very Broadband Seismographic Network [m
      [32mNI North-East Italy Broadband Network [m
      [32mOE Austrian Seismic Network [m
      [32mOX North-East Italy Seismic Network [m
      [32mRF Friuli Venezia Giulia Accelerometric Network, Italy [m
      [32mSI Sudtirol Network, Italy [m
      [32mSK Slovak National Network of Seismic Stations (SK-Net) [m
      [32mST Trentino Seismic Network, Italy [m
      [32mZ3_2015 AlpArray backbone temporary stations [m
      [32m
     Acknowledgment is extremely important for network operators
     providing open data. When preparing publications, please
     cite the data appropriately. The FDSN service at
    
         http://www.fdsn.org/networks/citation/?networks=BW+C4+CH+CR+CZ+GE+GR+GU+HU+IU+IV+MN+NI+OE+OX+RF+SI+SK+ST+Z3_2015
    
     provides a helpful guide based on available network
     Digital Object Identifiers.
      [m
     In case of problems with your request, plese use the contact form at
    
         http://www.orfeus-eu.org/organization/contact/form/?recipient=EIDA
```

**Comment by javiquinte** [comment]
Hi! I tried to check if there is actually data available using the new [availability WS](https://ws.resif.fr/resifws/availability/1/local=en), because the time window is very short. Unfortunately it was not possible and I received always a 204 error code. An example below
http://ws.resif.fr/fdsnws/availability/1/extent?net=Z3&sta=A265A&format=text
Any hint, @jschaeff ?

**Comment by jschaeff** [comment]
Hi @javiquinte , this is restricted data, for availability to return non-empty extent, you should add the "includerestricted" option (default=false).

furthermore, availability is not very quick, so asking the extent without start and end can hit timeout. We will work on it.

This request returns some results:
```
http "ws.resif.fr/fdsnws/availability/1/extent?net=Z3&sta=A164A&format=text&includerestricted=true&start=2017-08-23T07:29:00&end=2017-08-23T07:39:00" 
HTTP/1.1 200 OK
Connection: Keep-Alive
Content-Length: 318
Content-type: text/plain; charset=UTF-8
Date: Tue, 12 Nov 2019 10:03:35 GMT
Keep-Alive: timeout=5, max=100
Server: gunicorn/19.9.0

#Network Station Location Channel Quality SampleRate Earliest                    Latest                      Updated                     TimeSpans Restriction
Z3       A164A   00       HHZ     M       100.0      2017-08-23T07:29:00.000000Z 2017-08-23T07:39:00.000000Z 2018-02-26T09:07:10.146377Z 1         RESTRICTED
```
I'll check our logs to trace this request and report back here.

**Comment by jschaeff** [comment]
Can you provide the time when te request was issued by the user ? Could you invite the user in this issue ?

**Comment by javiquinte** [comment]
Hi @jbienkowski 
I think found out what the problem actually is! Your user is specifying strictly an **empty location**, while all french Z3 stations seem to have location 00.
Check for instance the availability for that time window with the link below

http://ws.resif.fr/fdsnws/availability/1/extent?net=Z3&cha=HHZ&format=text&start=2017-08-23T07:29:00&end=2017-08-23T07:39:00&includerestricted=true

```
#Network Station Location Channel Quality SampleRate Earliest                    Latest                      Updated                     TimeSpans Restriction
Z3       A151A   00       HHZ     M       100.0      2017-08-23T07:29:00.000000Z 2017-08-23T07:39:00.000000Z 2018-02-26T09:08:08.217529Z 1         RESTRICTED 
Z3       A152A   00       HHZ     M       100.0      2017-08-23T07:29:00.000000Z 2017-08-23T07:39:00.000000Z 2018-02-26T09:08:08.994956Z 1         RESTRICTED 
Z3       A153A   00       HHZ     M       100.0      2017-08-23T07:29:00.000000Z 2017-08-23T07:39:00.000000Z 2018-02-26T09:08:09.618595Z 1         RESTRICTED 
Z3       A154A   00       HHZ     M       100.0      2017-08-23T07:29:00.000000Z 2017-08-23T07:39:00.000000Z 2018-03-13T09:07:30.544873Z 1         RESTRICTED 
Z3       A155A   00       HHZ     M       100.0      2017-08-23T07:29:00.000000Z 2017-08-23T07:39:00.000000Z 2018-02-26T09:07:00.438391Z 1         RESTRICTED 
Z3       A156A   00       HHZ     M       100.0      2017-08-23T07:29:00.000000Z 2017-08-23T07:39:00.000000Z 2018-02-26T09:07:01.814257Z 1         RESTRICTED 
...many more
```

Can you contact your user and confirm if this is the problem?

Thanks!

**Comment by jschaeff** [comment]
Nice catch @javiquinte !

**Comment by manconia** [comment]
Hi @javiquinte  and all. Thanks for checking this, I am the user raising the issue.

I removed the location from my fdsn_fetch query (thanks for this catch!) and I now see the RESIF AlpArray temporary stations if I use net=Z3. However, if I call the virtual net=_ALPARRAY and try to get the .mseed files I actually get only data from 4 RESIF stations and none of the Z3...

Any idea?


-----
From the AlpArray website:

Virtual Network Code _ALPARRAY 
In order to identify and facilitate easy and rapid access to all AlpArray Seismic Network stations, we have created a virtual network for AlpArray - _ALPARRAY. This network code can be used using both the EIDA data access portal and scripts using fdsn_fetch or arclink_fetch.

The rules defining which stations are included in _ALPARRAY are:

- include all permanent stations that are within the 250km distance polygon AND providing HH? channel data AND  operational at any time between 1.1.2016 and 1.1.2020

- include all Z3 network code stations (temporary AlpArray stations)

There are a number of exceptions that are also included (whitelisted):
- G.ECH, RD.LOR, RD.ORIF, RD.PGF (stations only recording at 20sps)
- CR.BRJN, CR.KALN, CR.MOSL, CR.NVLJ, CR.OZLJ, CR.PTJ, CR.RIY, CR.UDBI, CR.ZAG (stations only recording at 50sps)
- CR.HVAR, CR.KIJV, CR.MORI, CR.STON (50sps AND outside polygon)
- HS.ERNH (station just outside polygon)
